pkg/loop/internal/ccip/test/onramp_test.go

```diff
@@ -39,19 +37,14 @@ func TestOnRampGRPC(t *testing.T) {
 	t.Parallel()
 	ctx := tests.Context(t)
 	// create a price registry server
-	port := freeport.GetOne(t)
-	addr := fmt.Sprintf("localhost:%d", port)
-	lis, err := net.Listen("tcp", addr)
-	require.NoError(t, err, "failed to listen on port %d", port)
-	t.Cleanup(func() { lis.Close() })
 	// we explicitly stop the server later, do not add a cleanup function here
-	testServer := grpc.NewServer()
+	lis, testServer, addr := tests.SetupServer(t)
```

Comment:
why do we need to return lis?

tests.SetupGRPCServer would be a better name

Reply:
Fair - I'm not sure how much I like it either. I followed a similar pattern to WSRPC tests.